l1-contracts/deploy-scripts/upgrade/default_upgrade/DefaultCTMUpgrade.s.sol

```diff
@@ -227,16 +233,62 @@ contract DefaultCTMUpgrade is Script, CTMUpgradeBase {
         if (toml.keyExists("$.chain_creation_params.evm_emulator_hash")) {
             chainCreationParams.evmEmulatorHash = toml.readBytes32("$.chain_creation_params.evm_emulator_hash");
         }
+    }
+
+
+    function initializeConfigForPatchUpgrade(
```

Comment:
I would probably create two contracts on top of DefaultCTMUpgrade, defaultPatchCTMUpgrade and defaultFullCTMUpgrade, and move getProposedPatchUpgrade as getProposedUpgrade into DefaultPatchCTMUpgrade, and move the current getProposedUpgrade to defaultFullCTMUpgrade. 

Similary for this line, just implement it as initializeConfig in defaultPatchCTMUpgrade